### PR TITLE
Fix Client sock to bind same port as that of server

### PR DESCRIPTION
**Pull request by ami-GS**
## Description

Fix of https://github.com/microsoft/msquic/issues/2396
SO_REUSEPORT is for server setting.
RemoteAddress == NULL is for server sock. The **or** condition was wrong

## Testing

Running locally.... and trying to run on CI

## Documentation

N/A

### CODE REVIEW

**Comment by ami-GS** [comment]
With this fix, any client cannot use SO_REUSEPORT at all.
The `ForceShare` is set in QuicConnStart based on Connection->State.ShareBinding, but RemoteServer param should be checked whether it is loopback?

File: src/platform/datapath_epoll.c [hunk 1]
```diff
@@ -946,9 +946,7 @@ QUIC_STATUS
 CxPlatSocketContextInitialize(
     _Inout_ CXPLAT_SOCKET_CONTEXT* SocketContext,
     _In_ const QUIC_ADDR* LocalAddress,
-    _In_ const QUIC_ADDR* RemoteAddress,
-    _In_ BOOLEAN ForceShare
-    )
+    _In_ const QUIC_ADDR* RemoteAddress)
```
Comment (nibanks) [inline src/platform/datapath_epoll.c:949]:
```suggestion
    _In_ const QUIC_ADDR* RemoteAddress
    )
```

**Review by nibanks — DISMISSED** [review]

**Comment by ami-GS** [comment]
This affects `DataPathTest/DataPathTest.UdpShareClientSocket/*` as there are two client sockets, which the second one use first one's localaddress.
Should we delete this test? because we don't expect this use case as you mentioned
https://github.com/microsoft/msquic/blob/fdd15b3d6f91e912c8212ad709edc7d5dcb03627/src/platform/unittest/DataPathTest.cpp#L917-L923

**Comment by nibanks** [comment]
> This affects `DataPathTest/DataPathTest.UdpShareClientSocket/*` as there are two client sockets, which the second one use first one's localaddress. Should we delete this test? because we don't expect this use case as you mentioned
> 
> https://github.com/microsoft/msquic/blob/fdd15b3d6f91e912c8212ad709edc7d5dcb03627/src/platform/unittest/DataPathTest.cpp#L917-L923

If we don't need the scenario for QUIC, then yes, go ahead and delete it.

**Comment by nibanks** [comment]
@ami-GS after looking at your most recent PR I realized why that code was there (I had forgot). We (MsQuic) support (only on Linux today) the ability to reuse the **local** address/port to connect to different remote addresses/ports for client connections. So, the platform layer was actually correct.

The real problem here is that the `QuicLibraryGetBinding` logic for `QUIC_SHARED_EPHEMERAL_WORKAROUND` isn't correctly preventing a new `PortUnspecified` client socket from matching an existing server socket. This PR should simply be abandoned (sorry!) and the correct fix made at the QUIC layer.

On a side note, I'm surprised to see we didn't have QUIC level tests that exercised the shared local ports for client sockets. I thought we did... Are you sure no QUIC layer tests were failing with this PR (I didn't check)?

**Comment by nibanks** [comment]
See https://github.com/microsoft/msquic/pull/3439.

**Pull request closed without merging**